ldk-server/src/main.rs

```diff
@@ -154,18 +156,44 @@ fn main() {
 								payment_id, payment_hash, amount_msat
 							);
 							let payment_id = payment_id.expect("PaymentId expected for ldk-server >=0.1");
-							upsert_payment_details(&event_node, Arc::clone(&paginated_store) as Arc<dyn PaginatedKVStore>, &payment_id);
+
+							publish_event_and_upsert_payment(&payment_id,
+								|payment| event_envelope::Event::PaymentReceived(events::PaymentReceived {
+                    payment: Some(payment.clone()),
```

Comment:
Why clone here if `payment` is passed by value?

Reply:
nit: Could have just not cloned in the closures, FWIW. Cloning at the call site would be better IMO given it is a conversion function and you needed the value again later.